app/services/segment/gbpr_api_request_service.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Segment
+  module GBPRApiRequestService
+    module_function
+
+    def call(request_body:, custom_headers: {})
+      return unless Features::SegmentDeletionConfig.enabled?
+      connection = Faraday.new(Features::SegmentDeletionConfig.config.uri) do |faraday|
```

Comment:
We could move the creation of the connection to another method and simplify a bit the call. In fact., if the service could be instantiated, we could even memoize the connection and reused it in case of consecutive request. WDYT?

Reply:
It sounds good, I will check and either do it or let you know conclusions 👍